changefeedccl: create kafka topics via api call

asg0451 opened the pull request:
Add support for creating kafka topics via api
call, instead of relying on the kafka cluster
having auto topic creation enabled. This can be
enabled with the changefeed option
`create_kafka_topics='yes'`. The option can also
be set to `'no'` to disable topic creation, or
`'auto'` to automatically create topics if they
don't exist (default) (previous behaviour).

Fixes: #155157

Release note (general change): The changefeed option
`create_kafka_topics` now supports the values `'yes'`,
`'no'`, and `'auto'` (default).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153045)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: create kafka topics via api call

Add support for creating kafka topics via api
call, instead of relying on the kafka cluster
having auto topic creation enabled. This can be
enabled with the changefeed option
`create_kafka_topics='yes'`. The option can also
be set to `'no'` to disable topic creation, or
`'auto'` to automatically create topics if they
don't exist (default) (previous behaviour).

Fixes: #155157

Release note (general change): The changefeed option
`create_kafka_topics` now supports the values `'yes'`,
`'no'`, and `'auto'` (default).

asg0451 committed:
fixup! changefeedccl: create kafka topics via api call
